plugins/system/userlogs/userlogs.php

```diff
@@ -129,474 +171,538 @@ public function onContentAfterSave($context, $article, $isNew)
-            $isNew_string = $isNew ? 'true' : 'false';
-            $message = '{"event":"onExtensionAfterSave","title":"'.$table->title.'","isNew":"'.$isNew_string.'"}';
-            $this->addLogsToDB($message, $context);
-        }
-    }
-
-    /**
-     * On Deleting extensions logging method
-     * Method is called when an extension is being deleted
-     *
-     * @param   string  $context    The extension
-     * @param   JTable     $table   DataBase Table object
-     *
-     * @return  void
-     *
-     * @since   3.7
-     */
-    public function onExtensionAfterDelete($context, $table)
-    {
-        if($this->checkLoggable($this->jinput->get('option')))
-        {
-            $isNew_string = $isNew ? 'true' : 'false';
-            $message = '{"event":"onExtensionAfterDelete","title":"'.$table->title.'"}';
-            $this->addLogsToDB($message, $context);
-        }
-    }
-    /**
+	public function onExtensionAfterUninstall($installer, $eid, $result)
+	{
+		$context = $this->app->input->get('option');
+
+		if ($this->checkLoggable($context))
+		{
+			$message = '{"event":"onExtensionAfterUninstall","extenstion_name":"' . $installer->get('manifest')->name
+				. '","extenstion_type":"' . $installer->get('manifest')->attributes()['type'] . '"}';
+			$this->addLogsToDb($message, $context);
+		}
+	}
+
+	/**
+	 * On updating extensions logging method
+	 * This method adds a record to #__user_logs contains (message, date, context, user)
+	 * Method is called when an extension is updated
+	 *
+	 * @param   JInstaller  $installer  Installer instance
+	 * @param   integer	 $eid		Extension id
+	 *
+	 * @return  void
+	 *
+	 * @since   3.7
+	 */
+	public function onExtensionAfterUpdate($installer, $eid)
+	{
+		$context = $this->app->input->get('option');
+
+		if ($this->checkLoggable($context))
+		{
+			$message = '{"event":"onExtensionAfterUpdate","extenstion_name":"' . $installer->get('manifest')->name .
+				'","extenstion_type":"' . $installer->get('manifest')->attributes()['type'] . '"}';
+			$this->addLogsToDb($message, $context);
+		}
+	}
+
+	/**
+	 * On Saving extensions logging method
+	 * Method is called when an extension is being saved
+	 *
+	 * @param   string   $context   The extension
+	 * @param   JTable   $table	 DataBase Table object
+	 * @param   boolean  $isNew	 If the extension is new or not
+	 *
+	 * @return  void
+	 *
+	 * @since   3.7
+	 */
+	public function onExtensionAfterSave($context, $table, $isNew)
+	{
+		if ($this->checkLoggable($this->app->input->get('option')))
+		{
+			$parameters = UserlogsHelper::getLogMessageParams($context);
+			$title_holder = "";
+			$title_type = "";
+
+			if ($parameters)
+			{
+				$title_holder = $content->get($parameters->title_holder);
+				$type_title = $parameters->type_title;
+			}
+
+			$isNew_string = $isNew ? 'true' : 'false';
+			$message = '{"title":"' . $table->get($parameters->title_holder) . '","isNew":"' . $isNew_string . '", "event":"onExtensionAfterSave",' .
```

Comment:
imo it would be nicer to build an Array or stdClass here..

```php
$message = array(
   title = $table.., 
   message->isNew = ..
); 
```

and then use:
`
json_encode($message)`